EIPS/eip-5805.md

```diff
@@ -69,7 +69,31 @@ This function is thus optional, and its absence should be considered as a marker
   inputs: []
   outputs:
     - name: timepoint
-      type: uint256
+      type: uint48
+```
+
+### CLOCK_MODE
+
+This function returns a string describing the operating mode the contract is operating on.
+
+- If operating using **block number**, then this function SHOULD be implemented and return `mode=blocknumber&from=default`.
+- If operating using **timestamp**, then this function SHOULD be implemented and return `mode=timestamp`.
+- If operating using any other mode, then this function MUST be implemented. In particular, if a contract on some L2 uses the blocknumber of the L1 chain it is anchored to (mainnet in our example), the mode should describe that as `mode=blocknumber&from=eip155:1`.
+
+This function is thus optional, and its absence should be considered as a marker of the contract operating using block number or timestamp, which can both be clearly idenfitied from the absence of this function. (This makes this EIP compatible with pre-existing voting contracts).
+
+Note that when operating using **block number**, the `clock()` is expected to returns the value given by the `number` opcode (`0x43`). In some cases this can be the block number of another chain (in arbitrum, opcode `0x43` returns the block number of the last recorded operation on the parent chain). A contract can use `from=default` to specify that the block number used is the one provided by the `number` opcode (`0x43`). If a more explicit description is needed, CAIP-2 blockchain id should be used, as shown in the example above.
```

Comment:
>as shown in the example above

Just as a note I don't think we should see this as an example, we should be fully specifying it.